### ISSUE TITLE
Declaration of TeamTNT\Scout\Engines\TNTSearchEngine::map(Laravel\Scout\Builder $builder, $results, $model) must be    compatible with Laravel\Scout\Engines\Engine::map($results, $model)

### ISSUE THREAD

**Issue by AxDSan** [teamtnt/laravel-scout-tntsearch-driver#177]
I'm trying to install PyroCMS when I execute the command `php artisan install` I follow through the installation up to the point where it is seeding the database... there it prompts me with this error message: 
![](https://i.imgur.com/0m1WY1s.png)

I couldn't decide if dropping this here directly to you guys, or if to open an issue at PyroCMS GitHub.

**Comment by stevesweets** [comment]
@AxDSan This rather looks like a version clash... TNTSearchEngine has been updated to work with the most recent scout version, which isn't being required by @pyrocms somehow;

**Comment by RyanThompson** [comment]
Our composer file should have prevented this from happening though. It's a breaking change in a stable branch:

We require Scout `3.0.*` and this package as `3.0.*`: https://github.com/anomalylabs/streams-platform/blob/cad59c497de575f671834895ddfec5f218342531/composer.json#L12-L29

So this is indeed breaking.

**Comment by RyanThompson** [comment]
@AxDSan this has been resolved for the time being by updating to Scout 5. A couple small changes in core were also included but the 1.3 and 1.4 platform branches have both been updated around this.

**Comment by AxDSan** [comment]
Amazing, thanks for looking into this!

**Comment by chssmith** [comment]
We are having this issue separate from pyrocms.

Upgrading to Scout 5 didn't seem to work, as it was expecting a _soft__deleted column which we  didn't have in our tables.

I hardcoded a requirement for laravel/scout 4.0.1, teamtnt/tntsearch 1.3.1, and teamtnt/laravel-scout-tntsearch-driver 3.0.6 in our composer.json file, and it's now back to functioning.  These were the versions we were using before things broke when we updated today.
